### PR TITLE
Added mentor profile model

### LINKED ISSUE #9: Model for mentor profile
#### Description
Dependent on #7 and #8 
Create the model for the mentor profile collection as defined in the [doc](https://docs.google.com/document/d/1DxmTTWx1o55y6PT6zevxzY_57uSXAF3UJhK1l3lXlm4/edit#)

#### Files
`/backend/api/models/[namehere].py`

#### Tasks
Include specific tasks in the order they need to be done in. Include links to specific lines of code where the task should happen at.
- [ ] List all the fields that are not related other models (education, videos)
- [ ] Embed or reference the other models in this one, depending on whether they are EmbeddedDocuments (embed it like Person has Email embedded) or Documents (reference it)

### PR DESCRIPTION
**Pull request by AzureQDragon**
## Status: 
:rocket: Ready

## Description :star2:
Added mentor profile model with all the necessary fields.
MongoDB API Reference: https://docs.mongoengine.org/apireference.html
Fixes #9 
 
## TODOs :star:

- [ ] Model Review

### CODE REVIEW

**Review by Leundai — CHANGES_REQUESTED** [review]
within __init__.py of the folder models make sure to also include MentorProfile in there. like so
image

**Review by Leundai — APPROVED** [review]

File: backend/api/models/MentorProfile.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    offers_group_appointments = BooleanField(required=True)
+    video_links = EmbeddedDocumentField(VideoLinks)
+
+    def __repr__(self):
+        return f"""<Mentor id:{self.uid} \n name: {self.name} 
+                \n professional title: {self.professional_title} 
+                \n linkedin: {self.linkedin} \n website: {self.website}
+                \n picture: {self.picture} \n biography: {self.biography} 
+                \n offers_in_person: {self.offers_in_person} 
+                \n offers_group_appointments: {self.offers_group_appointments}>"""
```
Comment (chyku) [inline backend/api/models/MentorProfile.py:30]: Run `black .` to format! Will be adding checks in later today for this

File: backend/api/models/__init__.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 from .base import db
 from .Email import Email
 from .Person import Person
-
-__all__ = ["db", "Email", "Person"]
+from .MentorProfile import MentorProfile
+__all__ = ["db", "Email", "Person", "MentorProfile"]
```
Comment (chyku) [inline backend/api/models/__init__.py:8]: there should be a newline between the imports and the definition of `__all__`

File: backend/api/models/MentorProfile.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+from api.core import Mixin
+from .base import db
+from api.models import Email
+from flask_mongoengine import Document
+from mongoengine import *
+
+class MentorProfile(Document, Mixin):
+    """"Mentor Profile Collection."""
+    
+    uid = "" # TBD later
```
Comment (chyku) [inline backend/api/models/MentorProfile.py:10]: make the comment a `# TODO: [thing you need to do]` for convention + description of the task!

File: backend/api/models/MentorProfile.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    name = StringField(required=True)
+    professional_title = StringField(required=True)
+    linkedin = StringField(required=True)
+    website = StringField(required=True)
+    picture = StringField(required=True)
+    education = EmbeddedDocumentField(Education)
+    languages = ListField(StringField(required=True))
+    specializations = ListField(StringField(required=True))
```
Comment (chyku) [inline backend/api/models/MentorProfile.py:18]: not sure how this exactly works with ListFields, but shouldn't the ListField itself be required, not the strings within it?

**Review by chyku — CHANGES_REQUESTED** [review]
lookin good!!

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "/bin/python3"
+}
```
Comment (chyku) [inline .vscode/settings.json:3]: this shouldn't be committed! Make sure you add this to `.gitignore` if it's not already

File: frontend/package-lock.json [hunk 6]
```diff
@@ -0,0 +1,1408 @@
+{
```
Comment (chyku) [inline frontend/package-lock.json:1]: Did you end up using npm instead of yarn to install packages? Not sure why you'd end up having a package-lock

**Review by chyku — CHANGES_REQUESTED** [review]
Files

**Review by chyku — APPROVED** [review]
looks :+1:

**Pull request merged**